packages/utils/props.ts

```diff
@@ -88,38 +116,31 @@ export function buildProp<
         }
       : undefined
 
-  type HasDefaultValue = Exclude<D, undefined> extends never ? false : true
-  type Type = PropType<
-    | (T extends PropWrapper<unknown>
-        ? T[typeof wrapperKey]
-        : [V] extends [never]
-        ? ResolvePropTypeWithReadonly<T>
-        : never)
-    | V
-    | C
-  >
-
   return {
     type: (type as any)?.[wrapperKey] || type,
     required: !!required,
     default: defaultValue,
     validator: _validator,
-  } as unknown as {
-    readonly type: Type
-    readonly required: R
-    readonly validator: typeof _validator
-  } & (R extends true
-    ? { readonly default?: undefined }
-    : {
-        readonly default: HasDefaultValue extends true
-          ? Exclude<
-              D extends Record<string, unknown> | Array<any> ? () => D : D,
-              undefined
-            >
-          : undefined
-      })
+  } as unknown as BuildPropReturn<T, D, R, V, C>
 }
 
+export const buildProps = <
+  O extends Record<string, BuildPropOption<any, any, any, any, any>>
+>(
+  options: O
+) =>
+  mapValues(options, (option) => buildProp(option)) as unknown as {
```

Comment:
Why are we not using `Object.values().map` for this one?

Reply:
We need keep the key